sound/soc/sdw_utils/soc_sdw_utils.c

```diff
@@ -1115,6 +1145,35 @@ int asoc_sdw_parse_sdw_endpoints(struct snd_soc_card *card,
 				if (dai_info->quirk && !(dai_info->quirk & ctx->mc_quirk))
 					continue;
 
+				dlc.dai_name = dai_info->dai_name;
+				codec_dai = snd_soc_find_dai_with_mutex(&dlc);
+				if (!codec_dai) {
+					dev_err(dev, "codec dai %s not registered yet\n",
+						dlc.dai_name);
+					return -EPROBE_DEFER;
+				}
+
+				slave = dev_to_sdw_dev(codec_dai->dev);
+				if (!slave)
+					return -EINVAL;
```

Comment:
Turns out there is one slight complication I had overlooked here as well, the codec device might not necessarily be the soundwire device. For example cs42l43 is an MFD so it is the parent of the codec device that is the soundwire slave, this situation would likely exist for the SDCA auxiliary device things as well.

Reply:
Thanks @charleskeepax for pointing this out. Let's figure out how to solve it. Maybe check up to 2 level parents?

Reply:
I have been playing around with using bus_find_device to search for the soundwire device instead I think that might be workable at the expense of perhaps being a little expensive.